tests/pytests/integration/cli/test_salt_call.py

```diff
@@ -256,12 +263,29 @@ def test_masterless_highstate(salt_call_cli):
     """
     test state.highstate in masterless mode
     """
-    destpath = os.path.join(RUNTIME_VARS.TMP, "testfile")
-    ret = salt_call_cli.run("--local", "state.highstate")
-    assert ret.exitcode == 0
-    state_run_dict = next(iter(ret.json.values()))
-    assert state_run_dict["result"] is True
-    assert state_run_dict["__id__"] == destpath
+    top_sls = """
+    base:
+      '*':
+        - core
+        """
+
+    core_state = """
+    {}/testfile:
+      file:
+        - managed
+        - source: salt://testfile
+        - makedirs: true
+        """.format(
+        RUNTIME_VARS.TMP
+    )
+
+    with temp_state_file("top.sls", top_sls), temp_state_file("core.sls", core_state):
```

Comment:
```suggestion
    with pytest.helpers.temp_file("top.sls", top_sls, base_env_state_tree_root_dir), pytest.helpers.temp_file("core.sls", core_state, base_env_state_tree_root_dir):
```
Be sure to add `base_env_state_tree_root_dir` as a test function argument.

The reason I suggest this is because I'd for use to move away from using `RUNTIME.*` since that was a hack because we didn't have fixtures in unittest :)